commands/update.go

```diff
@@ -0,0 +1,193 @@
+package commands
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/pathutil"
+	"github.com/pkg/errors"
+
+	"github.com/bitrise-io/go-utils/sliceutil"
+
+	"github.com/spf13/cobra"
+	"github.com/trapacska/repotool/storage"
+)
+
+var updateCmd = &cobra.Command{
+	Use:   "update",
+	Short: "Update repositories to follow repo URL change.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return updateRepos()
+	},
+	Args: func(cmd *cobra.Command, args []string) error {
+		if len(sourceFile) > 0 && len(sourceRepo) > 0 {
+			return fmt.Errorf("both --source-file and --source-repo cannot be set")
+		}
+
+		var authenticated bool
+		if err := storage.Read(func(table *storage.Table) {
+			authenticated = len(table.GithubAccessToken) > 0
+		}); err != nil {
+			return err
+		}
+
+		if !authenticated {
+			return fmt.Errorf("not authenticated")
+		}
+
+		return nil
+	},
+}
+
+func updateRepos() error {
+	var authToken string
+	var updatedRepos []string
+	var transferredRepos []storage.Repo
+
+	if err := storage.Read(func(table *storage.Table) {
+		authToken = table.GithubAccessToken
+		updatedRepos = table.UpdatedRepositories
+		transferredRepos = table.TransferredRepositories
+	}); err != nil {
+		return err
+	}
+
+	for _, repo := range transferredRepos {
+		from := repoOwner(repo.OriginalURL) + "/" + repoName(repo.OriginalURL)
+		to := repo.TargetOrg + "/" + repoName(repo.OriginalURL)
+		newURL := strings.Replace(repo.OriginalURL, from, to, -1)
+
+		if sliceutil.IsStringInSlice(newURL, updatedRepos) {
+			fmt.Println("Skip (already updated):", newURL)
+			continue
+		}
+
+		tempDir, err := pathutil.NormalizedOSTempDirPath("repo")
+		if err != nil {
+			log.Errorf("%s", err)
+			continue
+		}
+
+		if out, err := command.New("git", "clone", newURL, tempDir).RunAndReturnTrimmedCombinedOutput(); err != nil {
+			log.Errorf("%s", errors.Wrap(err, out))
+			continue
+		}
+
+		if out, err := command.New("git", "branch", "git-url-update").SetDir(tempDir).RunAndReturnTrimmedCombinedOutput(); err != nil {
+			log.Errorf("%s", errors.Wrap(err, out))
+			continue
+		}
+
+		if out, err := command.New("git", "checkout", "git-url-update").SetDir(tempDir).RunAndReturnTrimmedCombinedOutput(); err != nil {
+			log.Errorf("%s", errors.Wrap(err, out))
+			continue
+		}
+
+		if err := filepath.Walk(tempDir, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if info.IsDir() && info.Name() == ".git" {
```

Comment:
I think should not update inside vendor dir (or legacy deps) and Gopkg.lock (not sure about Gopkg.toml)

Reply:
I think almost anything need to be updated that has the wrong old path